qcodes/instrument_drivers/Shamrock/SR750.py

```diff
@@ -0,0 +1,320 @@
+        self.error_check(code, 'ShamrockGetWavelength')
+        return c_wavelength.value
+
+    def get_wavelength_limits(self, device, grating):
+        c_device = ctypes.c_int(device)
+        c_grating = ctypes.c_int(grating)
+        c_min = ctypes.c_float()
+        c_max = ctypes.c_float()
+        code = self.dll.ShamrockGetWavelengthLimits(c_device, c_grating, ctypes.byref(c_min), ctypes.byref(c_max))
+        self.error_check(code, 'ShamrockGetWavelengthLimits')
+        return c_min.value, c_max.value
+
+    def initialize(self):
+        code = self.dll.ShamrockInitialize("")
+        self.error_check(code, 'ShamrockInitialize')
+
+    def set_grating(self, device, grating):
+        c_device = ctypes.c_int(device)
+        c_grating = ctypes.c_int(grating)
+        code = self.dll.ShamrockSetGrating(c_device, c_grating)
+        self.error_check(code, 'ShamrockSetGrating')
+
+    def set_number_pixels(self, device, number_pixels):
+        c_device = ctypes.c_int(device)
+        c_number_pixels = ctypes.c_int(number_pixels)
+        code = self.dll.ShamrockSetNumberPixels(c_device, c_number_pixels)
+        self.error_check(code, 'ShamrockSetNumberPixels')
+
+    def set_pixel_width(self, device, width):
+        c_device = ctypes.c_int(device)
+        c_width = ctypes.c_float(width)
+        code = self.dll.ShamrockSetPixelWidth(c_device, c_width)
+        self.error_check(code, 'ShamrockSetPixelWidth')
+
+    def set_slit(self, device, width):
+        c_device = ctypes.c_int(device)
+        c_width = ctypes.c_float(width)
+        code = self.dll.ShamrockSetSlit(c_device, c_width)
+        self.error_check(code, 'ShamrockSetSlit')
+
+    def set_wavelength(self, device, wavelength):
+        c_device = ctypes.c_int(device)
+        c_wavelength = ctypes.c_float(wavelength)
+        code = self.dll.ShamrockSetWavelength(c_device, c_wavelength)
+        self.error_check(code, 'ShamrockSetWavelength')
+
+
+class Shamrock_SR750(Instrument):
+    """
+    Instrument driver for the Shamrock SR750 spectrometer.
+
+    Args:
+        name: Instrument name.
+        dll_path: Path to the ShamrockCIF.dll file. If not set, a default path is used.
+        device_id: ID for the desired spectrometer.
+        ccd_number_pixels: Number of pixels on the connected CCD.
+        ccd_pixel_width: Pixel width on the connected CCD.
+
+    Attributes:
+        ShamrockCIF: DLL wrapper for ShamrockCIF.dll
+        device_id: Spectrometer device ID.
+        serial_number: Serial number of the spectrometer.
+        number_gratings: Number of gratings on the spectrometer.
+    """
+
+    def __init__(self, name: str, dll_path: Optional[str] = None, device_id: Optional[int] = 0, ccd_number_pixels: Optional[int] = 1024, ccd_pixel_width: Optional[int] = 26, **kwargs):
+
+        super().__init__(name, **kwargs)
+
+        # link to dll
+        self.ShamrockCIF: ShamrockCIF = ShamrockCIF(dll_path=dll_path)
```

Comment:
For the future, it is very beneficial for testing the driver without the need for the actual DLL file to be present, to have an Optional argument for the "dll-wrapper" instance, so that if nothing is passed "ShamrockCIF" gets created as here, but if smth is passed then that is used instead of creating "ShamrockCIF" instance.